elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java

```diff
@@ -121,6 +124,49 @@ private ExprValue construct(String field, JsonNode value) {
     }
   }
 
+  /**
+   * Construct ExprValue from field and its value object. Throw exception if trying
+   * to construct from field of unsupported type.
+   *
+   * @param field   field name
+   * @param value   value object
+   * @return        ExprValue
+   */
+  public ExprValue construct(String field, Object value) {
+    if (value == null) {
+      return nullValue();
+    }
+
+    ExprType type = type(field);
+    if (type.equals(INTEGER)) {
+      return constructInteger((Integer) value);
+    } else if (type.equals(LONG)) {
+      return constructLong((Long) value);
+    } else if (type.equals(FLOAT)) {
+      return constructFloat((Float) value);
+    } else if (type.equals(DOUBLE)) {
+      return constructDouble((Double) value);
+    } else if (type.equals(STRING)) {
+      return constructString((String) value);
+    } else if (type.equals(BOOLEAN)) {
+      return constructBoolean((Boolean) value);
+    } else if (type.equals(TIMESTAMP)) {
+      if (value instanceof Number) {
+        return constructTimestamp((Long) value);
+      } else if (value instanceof Instant) {
+        return constructTimestamp((Instant) value);
+      } else {
+        return constructTimestamp(String.valueOf(value));
+      }
+    } else if (ES_TEXT.isCompatible(type)) {
+      return new ElasticsearchExprTextValue((String) value);
```

Comment:
Add some comments?

Reply:
Sure, will add a new `ExprValue` class to differentiate text and multi-field.